packages/integration-sdk-runtime/src/execution/executeIntegration.ts

```diff
@@ -256,9 +261,29 @@ export async function executeWithContext<
         resultsCallback,
       } = options;
 
-      const duplicateKeyTracker = new DuplicateKeyTracker(
-        config.normalizeGraphObjectKey,
-      );
+      let duplicateKeyTracker: DuplicateKeyTracker =
+        new InMemoryDuplicateKeyTracker(config.normalizeGraphObjectKey);
```

Comment:
Are we allocating memory unnecessarily by instantiating this regardless of whether it'll be used?
I'm thinking about this guy:
```
this.graphObjectKeyMap = new BigMap<string, string>(
      DUPLICATE_KEY_TRACKER_DEFAULT_MAP_KEY_SPACE,
    );
```